ORBIT/phases/design/electrical_export.py

```diff
@@ -492,32 +567,87 @@ def calc_topside_mass_and_cost(self):
+        # topside_fab_cost_rate = _design.get("topside_fab_cost_rate", 14500)
+        # topside_design_cost = _design.get("topside_design_cost", 4.5e6)
 
         self.topside_mass = (
             3.85 * (self.mpt_rating * self.num_mpt) / self.num_substations
             + 285
         )
-        self.topside_cost = (
-            self.topside_mass * topside_fab_cost_rate + topside_design_cost
-        )
+        if self.cable.cable_type == "HVDC-monopole":
+            self.topside_cost = _design.get("topside_design_cost", 294e6)
+        elif self.cable.cable_type == "HVDC-bipole":
+            self.topside_cost = _design.get("topside_design_cost", 476e6)
+        else:
+            self.topside_cost = _design.get("topside_design_cost", 107.3e6)
 
     def calc_onshore_cost(self):
-        """Minimum Cost of Onshore Substation Connection"""
+        """Minimum Cost of Onshore Substation Connection
+
+        Parameters
+        ----------
+        shunt_cost_rate : int | float
+        onshore_converter_cost: int | float
+        switchgear_cost: int | float
+        """
+
+        _design = self.config.get("onshore_substation_design", {})
+
+        _shunt_cost_rate = _design.get("shunt_cost_rate", 1.3e4)  # per cable
+        _switchgear_cost = _design.get("switchgear_cost", 9.33e6)  # per cable
+        _compensation_rate = _design.get(
+            "compensation_rate", 31.3e6
+        )  # per cable
+
+        self.onshore_shunt_reactor_cost = (
+            self.compensation * self.num_cables * _shunt_cost_rate
+        )
+
+        if self.cable.cable_type == "HVDC-monopole":
+            self.onshore_converter_cost = (
+                self.num_substations
+                * self._design.get("onshore_converter_cost", 157e6)
+            )
+            self.onshore_switchgear_cost = 0
+            self.onshore_construction = self.num_substations * _design.get(
+                "onshore_construction_rate", 87.3e6
+            )
+            self.onshore_compensation_cost = 0
+
+        elif self.cable.cable_type == "HVDC-bipole":
+            self.onshore_converter_cost = (
+                self.num_substations
+                * self._design.get("onshore_converter_cost", 350e6)
+            )
+            self.onshore_switchgear_cost = 0
+            self.onshore_construction = self.num_substations * _design.get(
+                "onshore_construction_rate", 100e6
+            )
+            self.onshore_compensation_cost = 0
+
+        else:
+            self.onshore_converter_cost = 0
+            self.onshore_switchgear_cost = self.num_cables * _switchgear_cost
+            self.onshore_construction = self.num_substations * _design.get(
+                "onshore_construction_rate", 5e6
+            )
+            self.onshore_compensation_cost = (
+                self.num_cables * _compensation_rate
+                + self.onshore_shunt_reactor_cost
+            )
```

Comment:
Something to think about for a 2.0 timeframe is how we could change default value setups like this to be a bit more streamlined and avoid this big `if/elif/else` blocks altogether.

Reply:
Yeah. This section has some overlap with `export.py`. There is some room to clean up this.